src/cmd/tools/m3ctl/client/checker.go

```diff
@@ -0,0 +1,21 @@
+package client
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"go.uber.org/zap"
+)
+
+func checkForAndHandleError(url string, resp *http.Response, zl *zap.Logger) error {
+	zl.Info(fmt.Sprintf("resp.StatusCode:%d:\n", resp.StatusCode))
+	if resp.StatusCode > 299 {
+		dat, _ := ioutil.ReadAll(resp.Body)
+		if dat != nil {
+			zl.Info(string(dat))
```

Comment:
Would suggest this being an error log and use fields:
```
if dat != nil {
  zl.Error("error response", zap.Error(fmt.Errorf("status %d", resp.StatusCode)), zap.String("url", url), zap.BytesString("response", dat))
} else {
  zl.Error("error response", zap.Error(fmt.Errorf("status %d", resp.StatusCode)), zap.String("url", url), zap.BytesString("response", dat))
}
```

Reply:
seems wrong that zap is pinned to f85c78b1dd998214c5f2138155b320a4a43fbe36 in glide.yaml.  I'll bump it to some version that contains BytesString.